packages/jaeger-ui/src/model/ddg/GraphModel/getPathElemHasher.test.js

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import GraphModel from './index';
+import transformDdgData from '../transformDdgData';
+import { EDdgDensity } from '../types';
+
+function makePayloadEntry(pairStr) {
+  const [service, operation] = pairStr.split(':');
+  return { service, operation };
+}
+
+const payload = `
+  a:0   b:0   focal:focal   c:0   d:0
+        b:0   focal:focal   c:0
+        c:0   focal:focal   b:0
+  a:0   c:0   focal:focal   b:0   d:0
+  a:1   b:0   focal:focal   c:0   d:1
+`
+  .trim()
+  .split('\n')
+  .map(line => ({
+    path: line
+      .trim()
+      .split(/\s+/g)
+      .map(makePayloadEntry),
+  }));
+
+const testTable = [
+  // showOp, density, number of expected vertices
+  [false, EDdgDensity.MostConcise, 5],
+  [true, EDdgDensity.MostConcise, 7],
+  [false, EDdgDensity.UpstreamVsDownstream, 7],
+  [true, EDdgDensity.UpstreamVsDownstream, 9],
+  [false, EDdgDensity.PreventPathEntanglement, 9],
+  [true, EDdgDensity.PreventPathEntanglement, 11],
+  [false, EDdgDensity.ExternalVsInternal, 13],
+  [true, EDdgDensity.ExternalVsInternal, 15],
```

Comment:
even the most verbose density setting should include some condensing. i.e.: the hashed value isn't just `map.size`. i'd add the row `a:1   b:0   focal:focal   c:0   d:0` to the payload which shouldn't cause any expected counts to change.

Reply:
also, in conjunction with [this PR](https://github.com/jaegertracing/jaeger-ui/pull/441/commits/6d9430cc7e6f8ca5d8c28de243df22564477592d) it'd be nice to see different numbers for all cases just to be sure nothing is being oddly crossed

Reply:
I'll update this test table on that branch to avoid conflicts